nilearn/decomposition/tests/test_canica.py

```diff
@@ -67,5 +80,30 @@ def test_canica_square_img():
     # Smoke test to make sure an error is raised when no data is passed.
     assert_raises(TypeError, canica.fit)
 
+
+def test_component_sign():
+    # BF for issue #570
+
+    # make data
+    data, mask_img, components, rng = _make_canica_test_data(n_subjects=2)
+
+    # fill with -1's
+    components[:] = -1
+
+    # have some +1's (so things are not degenerate), but still have more -1's
+    # than +1's
+    for mp in components:
+        mp[rng.randn(*mp.shape) > .8] *= -.5
+        assert_less_than(mp.max(), -mp.min())
```

Comment:
I don't understand the few lines above: what are they there for? It seems that they are not used.